cpp/include/ucxx/worker.h

```diff
@@ -643,6 +674,56 @@ class Worker : public Component {
   std::shared_ptr<Listener> createListener(uint16_t port,
                                            ucp_listener_conn_callback_t callback,
                                            void* callbackArgs);
+
+  /**
+   * @brief Register allocator for active messages.
+   *
+   * Register a new allocator for active messages. By default, only one allocator is defined
+   * for host memory (`UCS_MEMORY_TYPE_HOST`), and is used as a fallback when an allocator
+   * for the source's memory type is unavailable. In many circumstances relying exclusively
+   * on the host allocator is undesirable, for example when transferring CUDA buffers the
+   * destination is always going to be a host buffer and prevent the use of transports such
+   * as NVLink or InfiniBand+GPUDirectRDMA. For that reason it's important that the user
+   * defines those allocators that are important for the application.
+   *
+   * If the `memoryType` has already been registered, the previous allocator will be
+   * replaced by the new one. Be careful when doing this after transfers have started, there
+   * are no guarantees that inflight messages have not already been allocated with the old
+   * allocator for that type.
+   *
```

Comment:
Do we want to add some text here about how the data received by active messages must be deallocated using the allocator's "matching" `free` implementation? E.g. I see that the `HostBuffer` implementation uses `malloc`/`free` pairs, but it could equally use `new`/`delete`. (This is only _really_ relevant when doing `HostBuffer.release` I guess).

Reply:
I haven't thought about this, but given the object the user will ultimately get via `request->getRecvBuffer()` is a `Buffer`, I would hope [the user reads `Buffer::release()` docs](https://github.com/rapidsai/ucxx/blob/95f0e4280488daec3ecdd1870adba587ecea1f83/cpp/include/ucxx/buffer.h#L122-L125) which writes that. Would you like to be redundant here? To me the danger of redundancy is in the event of changes in the behavior of `Buffer` and missing to update the relevant bits here too, perhaps the ideal case would be to have a cross-ref instead, but I haven't done that in this repo yet.